src/codeflare_sdk/cluster/cluster.py

```diff
@@ -94,6 +94,7 @@ def up(self):
         Applies the AppWrapper yaml, pushing the resource request onto
         the MCAD queue.
         """
+        self.config.auth.login()
```

Comment:
Do we want to log in when we do cluster.up(), and logout with cluster.down()? Or do we want to wrap each call (up, down, etc) with a login at the beginning, and a logout at the end? The latter is what I thought we were originally going for, but I could see reasons why we may want to do this instead. Curious on your takes @atinsood @MichaelClifford

Reply:
In my opinion, I don't think that we want to login and out on every single call. This is not how a user would naturally interact with a cluster, and I'm not sure of the benefits of doing it that way. 

Also, on logout() your token can expire, which would break subsequent login in calls. 

It seems reasonable to me that you would want to login at the start of a sessions (on cluster.up()) and logout at the end (on cluster.down())  

Interested to hear other takes though :smile:

Reply:
Oh the expiration is an excellent point, I hadn't thought about that. I think your implementation is probably the correct way after all